src/main/java/juke/command/AddCommand.java

```diff
@@ -0,0 +1,112 @@
+package juke.command;
+
+import juke.exception.JukeInvalidParameterException;
+import juke.exception.JukeMissingArgumentException;
+import juke.exception.JukeParseException;
+import juke.task.Deadline;
+import juke.task.Event;
+import juke.task.TaskType;
+import juke.task.Todo;
+
+public class AddCommand extends Command {
+    private TaskType type;
+    
+    public AddCommand(TaskType type) {
+        this.type = type;
+    }
+    
+    @Override
+    public Command checkParametersAndArguments() {
+        switch (this.type) {
+        case EVENT:
+            if (this.hasParameter("at")) {
+                if (this.hasArgument("at")) {
+                
+                } else {
+                    this.result = Result.error(new JukeMissingArgumentException(this.type.getCommandName()));
+                    return this;
+                }
+            } else {
+                this.result = Result.error(new JukeMissingArgumentException(this.type.getCommandName()));
+                return this;
+            }
+            break;
+        case DEADLINE:
+            if (this.hasParameter("by")) {
+                if (this.hasArgument("by")) {
+            
+                } else {
+                    this.result = Result.error(new JukeMissingArgumentException(this.type.getCommandName()));
+                    return this;
+                }
+            } else {
+                this.result = Result.error(new JukeMissingArgumentException(this.type.getCommandName()));
+                return this;
+            }
+            break;
+        }
+        for (String param : this.paramArgs.keySet()) {
+            if (!this.isDefaultParameter(param)) {
+                switch (this.type) {
+                case EVENT:
+                    if (!param.equals("at")) {
+                        this.result = Result.error(new JukeInvalidParameterException(param));
+                        return this;
+                    }
+                    break;
+                case DEADLINE:
+                    if (!param.equals("by")) {
+                        this.result = Result.error(new JukeInvalidParameterException(param));
+                        return this;
+                    }
+                    break;
+                default:
+                    this.result = Result.error(new JukeInvalidParameterException(param));
+                    return this;
+                }
+            }
+        }
+        if (!this.hasDefaultArgument()) {
+            this.result = Result.error(new JukeMissingArgumentException(this.type.getCommandName()));
+            return this;
+        }
+        return this;
+    }
```

Comment:
"Result.error(new JukeMissingArgumentException(this.type.getCommandName()))" is repeated a lot, consider create a new method to fetch the result with only "this.type.getCommandName()" as the input?